backbone/src/main/java/org/researchstack/backbone/ui/step/layout/StepLayout.java

```diff
@@ -1,13 +1,23 @@
 package org.researchstack.backbone.ui.step.layout;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.view.View;
 
 import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.result.TaskResult;
 import org.researchstack.backbone.step.Step;
 import org.researchstack.backbone.ui.callbacks.StepCallbacks;
 
 public interface StepLayout
 {
-    void initialize(Step step, StepResult result);
+    /**
+     * @param step Step to be related to this StepLayout
+     * @param result the StepResult for this step, if one already exists
+     * @param taskResult The TaskResult object, if this StepLayout belongs to a Task
+     */
+    void initialize(@NonNull Step step,
+                    @Nullable StepResult result,
+                    @Nullable TaskResult taskResult);
```

Comment:
From what I can gather, Step comes from Task (getStepAfterStep, etc), so I'm wondering if the step being passed in could already have the relevant data set on it. Passing taskResult seems to ask a step to understand what a Task is, and what other Steps are, and so far it looks like Task knows about Step, not as much the other way around.

Another thing I'm thinking is that StepLayout has a callback, but a PR also recently added a callback on Task. Task could then be responsible for moving data around between steps.

I'll definitely defer to you guys since you have better context on what the use cases are and what RK does